appIcons.js

```diff
@@ -1228,11 +1228,14 @@ function getInterestingWindows(windows, monitorIndex) {
     let settings = Docking.DockManager.settings;
 
     // When using workspace isolation, we filter out windows
-    // that are not in the current workspace
-    if (settings.get_boolean('isolate-workspaces'))
+    // that are neither in the current workspace nor marked urgent
+    if (settings.get_boolean('isolate-workspaces')) {
+		const showUrgent = settings.get_boolean('show-urgent-windows');
         windows = windows.filter(function(w) {
-            return w.get_workspace().index() == global.workspace_manager.get_active_workspace_index();
+            return w.get_workspace().index() == global.workspace_manager.get_active_workspace_index()
```

Comment:
Since you're changing this, maybe you can move the `global.workspace_manager.get_active_workspace_index()` call to the outside scope? Going to C calls when not needed is better to be avoided, and repeating it for each window it's just uneeded.